src/Service/Indexer.php

```diff
@@ -0,0 +1,178 @@
+                    $qb->range($offset, $limit);
+                }
+                $qb->sort('nid', 'DESC');
+            },
+        ];
+    }
+
+    protected function getStorages()
+    {
+        return [
+            $this->entityTypeManager->getStorage('node'),
+            $this->entityTypeManager->getStorage('taxonomy_term'),
+        ];
+    }
+
+    public function queueAll($from, $limit, $offset)
+    {
+        $conditions = $this->getConditions($from, $limit, $offset);
+
+        foreach ($conditions as $et => $condition) {
+            printf("Queuing entities of type %s\n", $et);
+
+            $storage = $this->entityTypeManager->getStorage($et);
+            $qb = $storage->getQuery();
+            $condition($qb);
+            $ids = $qb->execute();
+
+            $total = count($ids);
+            $c = 0;
+
+            foreach (array_chunk($ids, 50) as $chunk) {
+                $this->resetCaches();
+                foreach ($chunk as $id) {
+                    $c++;
+                    $entity = $storage->load($id);
+                    printf(
+                        "\033[0G\033[K%03d/%03d",
+                        $id,
+                        $total
+                    );
+
+                    if (!$entity instanceof DocumentInterface) {
+                        continue;
+                    }
+
+                    wmsearch_queue($entity, true);
+                }
+            }
+
+            echo "\n[done]\n";
+        }
+    }
+
+    public function purge()
+    {
+        $this->index->deleteIndex();
+        $this->index->createIndex();
+    }
+
+    private function indexEntity(EntityInterface $entity)
+    {
+        static $fails;
+
+        $retries = 3;
+
+        if (!isset($fails)) {
+            $fails = 0;
+        }
+
+        try {
+            if ($entity instanceof TranslatableInterface) {
+                $this->indexTranslations($entity);
+                return;
+            }
+
+            $this->indexDocument($entity);
+        } catch (GuzzleException $e) {
+            $fails++;
+            if ($fails >= $retries) {
+                throw $e;
+            }
+            dump($e->getMessage());
+            sleep(1);
+            $this->indexEntity($entity);
+            return;
+        }
+        $fails = 0;
+    }
+
+    private function indexTranslations(TranslatableInterface $entity)
+    {
+        foreach ($entity->getTranslationLanguages() as $langId => $_) {
+            $translation = $entity->getTranslation($langId);
+            if ($translation instanceof Node && !$translation->isPublished()) {
+                continue;
+            }
+            $this->indexDocument($translation);
+        }
+    }
+
+    private function indexDocument($entity)
+    {
+        if (!$entity instanceof DocumentInterface) {
+            return;
+        }
+
+        $data['types'] = $entity->getElasticTypes();
+
+        // Separate 'related' type, we'll handle it differently
```

Comment:
Related code hoort IMO echt niet in wmsearch.

Wmsearch is gwn `page` mapping.

Reply:
Fixed